### PR TITLE
[NOT AZURE] Azure Stack Storage Admin SDK

### PR DESCRIPTION
**Pull request by deathly809**
## Description

This is **NOT** a part of Azure.  This is the **Azure Stack** storage admin endpoints specifications.

https://github.com/Azure/azure-rest-api-specs/pull/2871

---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [X] Please add REST spec PR link to the SDK PR
- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [X] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [X] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [X] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [X] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/AzureStack/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj [hunk 1]
```diff
@@ -24,4 +24,5 @@
     <!-- Please do not move/edit code below this line -->
     <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
     <!-- Please do not move/edit code above this line -->
+
```
Comment (dsgouda) [inline src/AzureStack/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: Please undo these changes
in other files too

File: src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+//
+//      Major Version
+//      Minor Version
+//      Build Number
+//      Revision
+//
+// You can specify all the values or you can default the Build and Revision Numbers
+// by using the '*' as shown below:
+// [assembly: AssemblyVersion("1.0.*")]
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (dsgouda) [inline src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs]: Start versioning from `0.1.0.0`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good for the most part
Please checkout files with newline additions only

File: src/AzureStack/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj (hunk 1 above)
Comment (deathly809) [reply src/AzureStack/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: So I should not ever include these?  Some people tell me to add it and others don't.  Please point me towards a csproj that I should match and I will make them all have the same structure.

Comment (dsgouda) [reply src/AzureStack/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: The only thing that changed in this file is adding a newline (not necessary), everything else is okay, all you need to do is revert this change

File: src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (deathly809) [reply src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs]: So all the admin SDKs should start from 0.1.0.0?  Some are 1.0.0.0 and others are 0.1.0.0 we are getting inconsistent requests on what to put here.

**Comment by deathly809** [comment]
@dsgouda I have some questions.  Please see responses.

File: src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs (hunk 2 above)
Comment (dsgouda) [reply src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs]: @shahabhijeet what's the best advise here

**Comment by dsgouda** [comment]
@deathly809 responded

File: src/AzureStack/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj (hunk 1 above)
Comment (deathly809) [reply src/AzureStack/FabricAdmin/Fabric.Admin/Microsoft.AzureStack.Management.Fabric.Admin.csproj]: Done, will push when I find out if I should do 0.1.0.0 or 1.0.0.0 in the assembly.

**Comment by deathly809** [comment]
@dsgouda How does everything look?

File: src/AzureStack/GalleryAdmin/Gallery.Admin/Microsoft.AzureStack.Management.Gallery.Admin.csproj [hunk 3]
```diff
@@ -1,4 +1,5 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk">
+
```
Comment (dsgouda) [inline src/AzureStack/GalleryAdmin/Gallery.Admin/Microsoft.AzureStack.Management.Gallery.Admin.csproj]: nit: no useful changes here

File: src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+// Setting ComVisible to false makes the types in this assembly not visible
+// to COM components.  If you need to access a type in this assembly from
+// COM, set the ComVisible attribute to true on that type.
+[assembly: ComVisible(false)]
+
+// The following GUID is for the ID of the typelib if this project is exposed to COM
+[assembly: Guid("5f30ad46-8f37-4416-8eae-5b56a1471926")]
+
+[assembly: AssemblyVersion("1.0.0.0")]
```
Comment (dsgouda) [inline src/AzureStack/StorageAdmin/Storage.Admin/Properties/AssemblyInfo.cs]: AssemblyVersion needs to be `0.1.0.0`

**Review by dsgouda — CHANGES_REQUESTED** [review]
Looks good 
just fix the one assembly.cs and I shall merge

**Comment by dsgouda** [comment]
@deathly809 please resolve conflicts and checked out files with unnecessary changes

File: src/AzureStack/GalleryAdmin/Gallery.Admin/Microsoft.AzureStack.Management.Gallery.Admin.csproj (hunk 3 above)
Comment (deathly809) [reply src/AzureStack/GalleryAdmin/Gallery.Admin/Microsoft.AzureStack.Management.Gallery.Admin.csproj]: Leaving so that all the Admin csproj files have the exact same structure.

**Comment by deathly809** [comment]
@dsgouda Pulled from upstream.  Leaving space so that all csproj files look identical for admin.

**Review by dsgouda — APPROVED** [review]

**Comment by deathly809** [comment]
@dsgouda CI passed :)
